cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -450,177 +450,178 @@ message(STATUS "Found hdfs.h at: " ${HDFS_H_PATH})
 
 include_directories(SYSTEM "${HADOOP_HOME}/include")
 
+if (ARROW_WITH_ZLIB)
 # ----------------------------------------------------------------------
 # ZLIB
 
-if (NOT ARROW_ZLIB_VENDORED)
-  find_package(ZLIB)
-endif()
-
-if (NOT ZLIB_FOUND)
-  set(ZLIB_PREFIX "${CMAKE_CURRENT_BINARY_DIR}/zlib_ep/src/zlib_ep-install")
-  set(ZLIB_HOME "${ZLIB_PREFIX}")
-  set(ZLIB_INCLUDE_DIR "${ZLIB_PREFIX}/include")
-  if (MSVC)
-    if (${UPPERCASE_BUILD_TYPE} STREQUAL "DEBUG")
-      set(ZLIB_STATIC_LIB_NAME zlibstaticd.lib)
+  if("${ZLIB_HOME}" STREQUAL "")
```

Comment:
@wesm , by default `ZLIB_HOME` is not set, so ZLib will be built from sources similar like we do with`ARROW_ZLIB_VENDORED` by default. I might miss something, but with my best understanding the logic should be the same as before, but instead of setting of `ARROW_ZLIB_VENDORED` to OFF, it will be needed to set only `ZLIB_HOME` to path with zlib library.
Also it seems FindZLIB.cmake with current implementation in ThirdpartyToolchain.cmake will not search in default paths at all, since `FindZLIB.cmake` is called only if `ZLIB_HOME` is set. So some default distribution of zlib shouldn't be taken because of that, hopefully.
In next PR will create additional Appveyor build job steps to verify that if ARROW_BUILD_TOOLCHAIN is that, the found zlib lib is found from toolchain's path.